src/compiler/crystal/config.cr

```diff
@@ -14,11 +14,13 @@ module Crystal
 
     def self.description
       version, sha = version_and_sha
-      if sha
```

Comment:
Indeed, yet `Crystal::Config.version` doesn't include _sha_ and compile date making it less detailed.

Some examples of returned `--version` output for other popular software:

- Ruby
  ```
  ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin15]
  ```
- Node.js
  ```
  v8.1.2
  ```
- Rust
  ```
  rustc 1.19.0
  ```
- Bundler
  ```
  Bundler version 1.15.3
  ```
- Bower/Yarn/NPM/RubyGems
  ```
  1.8.0
  ```

They're all one-liners.
Multi-line output is much less common, although it happens too:

- Java
  ```
  java version "1.8.0_112"
  Java(TM) SE Runtime Environment (build 1.8.0_112-b16)
  Java HotSpot(TM) 64-Bit Server VM (build 25.112-b16, mixed mode)
  ```
- Swift
  ```
  Apple Swift version 3.0.2 (swiftlang-800.0.63 clang-800.0.42.1)
  Target: x86_64-apple-macosx10.9
  ```

I'd still argue that oneline output is easier to read and parse, it also has wider usage rates.